lib/layer.js

```diff
@@ -66,8 +66,9 @@ Layer.prototype.handle_error = function handle_error(error, req, res, next) {
   }
 
   try {
-    fn(error, req, res, next)
+    fn(error, req, res, once(next))
   } catch (err) {
+    if (err === 'once') { throw new Error('cannot call `next` more than once') }
```

Comment:
We definitely do not want to ever purposely throw something that is un-catachable in some way.